ibm_db2/datadog_checks/ibm_db2/ibm_db2.py

```diff
@@ -63,21 +63,12 @@ def __init__(self, name, init_config, instances):
         )
 
     def check(self, instance):
-        connection = self.get_connection()
-        if connection is None:
-            self.service_check(
-                self.SERVICE_CHECK_CONNECT,
-                self.CRITICAL,
-                tags=self._tags,
-                message="Unable to create new connection to database: {}".format(self._db),
-            )
-            return
+        if self._conn is None:
```

Comment:
I think we should keep the service check calls outside of the `get_connection()` call. If `self._conn` is always a valid connection in check run >1, then `get_connection()` would never be called, and no `can_connect` service checks would be emitted, OK or CRITICAL.

Reply:
Good catch